pkg/statussyncer/status/controller/placementbinding_controller.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/types"
+	policyv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type placementBindingController struct {
+	client   client.Client
+	log      logr.Logger
+	areEqual func(client.Object, client.Object) bool
+}
+
+func (c *placementBindingController) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	reqLogger := c.log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+
+	instance := &policyv1.PlacementBinding{}
+	err := c.client.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		reqLogger.Error(err, "Reconciliation failed")
+		return ctrl.Result{}, err
+	}
+
+	for _, subject := range instance.Subjects {
+		if subject.APIGroup == policyv1.SchemeGroupVersion.Group &&
+			subject.Kind == policyv1.Kind {
+			err := c.updatePolicyStatus(ctx, subject.Name, request.Namespace,
+				instance.PlacementRef.Name, instance.Name)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	reqLogger.Info("Reconciliation complete.")
+
+	return ctrl.Result{}, nil
+}
+
+func (c *placementBindingController) updatePolicyStatus(ctx context.Context, name, namespace,
+	placementRule, placementBinding string) error {
+
+	policy := &policyv1.Policy{}
+	err := c.client.Get(ctx, types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, policy)
+	if err != nil {
+		return fmt.Errorf("failed to get policy: %w", err)
+	}
+
+	var exists bool
+	//TODO: need to handle placement in future
+	for _, placement := range policy.Status.Placement {
+		if placement.PlacementRule == placementRule {
+			exists = true
+			break
+		}
+	}
+
+	if !exists {
+		originalPolicy := policy.DeepCopy()
+
+		policy.Status.Placement = append(policy.Status.Placement, &policyv1.Placement{
+			PlacementBinding: placementBinding,
+			PlacementRule:    placementRule,
+		})
```

Comment:
should first check if they entry already exists in the placement section.
otherwise, we may get multiple duplicated entries.
(e.g, when binding is updated we might add duplicated entries)

Reply:
```
for _, placement := range policy.Status.Placement {
		if placement.PlacementRule == placementRule {
			exists = true
			break
		}
	}
```
here checked if the entry exists or not. if no, then update.